ch_tools/common/clickhouse/config/clickhouse.py

```diff
@@ -67,27 +65,47 @@ def load(try_preprocessed=True):
 
         # Otherwise load all server config files and perform manual merge and processing
         config = _load_config(CLICKHOUSE_SERVER_MAIN_CONFIG_PATH)
-        deep_merge(config, _load_config(CLICKHOUSE_SERVER_CLUSTER_CONFIG_PATH))
-        for file in os.listdir(CLICKHOUSE_SERVER_CONFIGD_PATH):
-            deep_merge(
-                config, _load_config(os.path.join(CLICKHOUSE_SERVER_CONFIGD_PATH, file))
-            )
+        if os.path.exists(CLICKHOUSE_SERVER_CONFIGD_PATH):
+            for file in os.listdir(CLICKHOUSE_SERVER_CONFIGD_PATH):
+                _merge_configs(
+                    config,
+                    _load_config(os.path.join(CLICKHOUSE_SERVER_CONFIGD_PATH, file)),
+                )
 
         # Process includes
-        root_key = next(iter(config))
-        root_section = config[root_key]
-        for key, config_section in root_section.copy().items():
-            if not isinstance(config_section, dict):
-                continue
+        root_section = first_value(config)
+        include_file = root_section.get("include_from")
+        if include_file:
+            include_config = first_value(_load_config(include_file))
+            _apply_config_directives(root_section, include_config)
 
-            include = config_section.get("@incl")
-            if not include:
-                continue
+        return ClickhouseConfig(config, preprocessed=False)
 
-            if include != key and include in root_section:
-                root_section[key] = root_section[include]
-                del root_section[include]
 
-            del config_section["@incl"]
+def _merge_configs(main_config, additional_config):
+    for key, value in additional_config.items():
+        if key not in main_config:
+            main_config[key] = value
+            continue
 
-        return ClickhouseConfig(config, preprocessed=False)
+        if isinstance(main_config[key], dict):
+            if isinstance(value, dict):
+                _merge_configs(main_config[key], value)
+            elif value is not None:
+                main_config[key] = value
+            continue
+
+        main_config[key] = value
+
+
+def _apply_config_directives(config_section, include_config):
+    for key, item in config_section.items():
+        if not isinstance(item, dict):
+            continue
+
+        include = item.get("@incl")
```

Comment:
Do you mean elements with `incl` attribute and some stuff within?

```
<include incl="...">
    <tag1>value1</tag1>
    <tag2>value2</tag2>
</include>
```

Don't consider this case. Didn't plan to support all ClickHouse config features. Just cover several features required for https://a.yandex-team.ru/review/5133943

Elements in the form `<include incl="..."/>` should work fine.

Reply:
I meant if it is correct to do `config_section[key] = include_config[include]` for `<include incl="other_key"` element because it should be replaced entirely and we shuld eventually get `<other_key>...</other_key>` in the config
Hm.. And how does `xmltodict` process multiple `include` nodes on the same level ?